app/src/main/java/com/vci/vectorcamapp/incomplete_session/presentation/IncompleteSessionScreen.kt

```diff
@@ -23,8 +30,58 @@ fun IncompleteSessionScreen(
         ) { session ->
             IncompleteSessionCard(
                 session = session,
-                onClick = { onAction(IncompleteSessionAction.ResumeSession(session.localId)) }
+                onClick = { onAction(IncompleteSessionAction.ResumeSession(session.localId)) },
+                onDelete = { onAction(IncompleteSessionAction.DeleteSession(session.localId)) }
             )
         }
     }
+
+    if (state.deleteDialogSessionId != null) {
+        AlertDialog(
+            onDismissRequest = {
+                onAction(IncompleteSessionAction.DismissDeleteDialog)
+            },
+            title = {
+                Text(
+                    text = "Delete Session?",
+                    style = MaterialTheme.typography.headlineMedium,
+                    color = MaterialTheme.colors.textPrimary
+                )
+            },
+            text = {
+                Text(
+                    text = "This will permanently delete the session and all associated images from your device. This action cannot be undone.",
+                    style = MaterialTheme.typography.bodyMedium,
+                    color = MaterialTheme.colors.textSecondary
+                )
+            },
+            confirmButton = {
+                Button(
+                    onClick = {
+                        onAction(IncompleteSessionAction.ConfirmDeleteSession(state.deleteDialogSessionId!!))
```

Comment:
Do you need the !! here? You already have a check above this to check if the ID is null. !! is considered bad practice.